pkg/utils/probe_helper.go

```diff
@@ -0,0 +1,62 @@
+package utils
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/types"
+	"open-cluster-management.io/addon-framework/pkg/agent"
+	workapiv1 "open-cluster-management.io/api/work/v1"
+)
+
+// DeploymentProber is to check the addon status based on status
+// of the agent deployment status
+type DeploymentProber struct {
+	deployments []types.NamespacedName
+}
+
+func NewDeploymentProber(deployments ...types.NamespacedName) *agent.HealthProber {
+	return &agent.HealthProber{
+		Type: agent.HealthProberTypeWork,
+		WorkProber: &DeploymentProber{
+			deployments: deployments,
+		},
+	}
+}
+
+func (d *DeploymentProber) ProbeFields() []agent.ProbeField {
+	probeFields := []agent.ProbeField{}
+	for _, deploy := range d.deployments {
+		probeFields = append(probeFields, agent.ProbeField{
+			ResourceIdentifier: workapiv1.ResourceIdentifier{
+				Group:     "apps",
+				Resource:  "deployments",
+				Name:      deploy.Name,
+				Namespace: deploy.Namespace,
+			},
+			ProbeRules: []workapiv1.FeedbackRule{
+				{
+					Type: workapiv1.WellKnownStatusType,
+				},
+			},
+		})
+	}
+	return probeFields
+}
+
+func (d *DeploymentProber) HealthCheck(identifier workapiv1.ResourceIdentifier, result workapiv1.StatusFeedbackResult) error {
+	if len(result.Values) == 0 {
```

Comment:
returning a non-nil error means the health-check controller will retry in the default backoff strategy which has a milli-sec backoff in the first 5-ish attempts IIRC, while the default interval of refreshing the feedback values is 30s. i think 0-length feedbacks can be somewhat reasonable when the workloads are applied for the first time (it may take a short period for the controller to process), so the health controller should better retry the health-check in a more "lazy" strategy e.g. constantly a 10-sec interval. otherwise the error log will spam.

Reply:
yeah, this is a good point.

Reply:
we check the len of values in controller. if there is no value returned, set the condition to unknown. I think we may not need resync, since work status update will trigger another check on this.